includes/class-newspack-ads-scaip.php

```diff
@@ -20,12 +20,28 @@ class Newspack_Ads_SCAIP {
 		'min_paragraphs' => 'scaip_settings_min_paragraphs',
 	);
 
+	// Default amount of ad insertions determined by SCAIP.
+	const DEFAULT_REPETITIONS = 2;
+
+	// Name of the hook to be created for the custom placement.
+	const HOOK_NAME = 'newspack_ads_scaip_placement_%s';
+
 	/**
 	 * Initialize SCAIP Hooks.
 	 */
 	public static function init() {
+		// Settings hooks.
 		add_filter( 'newspack_ads_settings_list', array( __CLASS__, 'add_settings' ) );
 		add_filter( 'newspack_ads_setting_option_name', array( __CLASS__, 'map_option_name' ), 10, 2 );
+
+		// Placements hooks.
+		add_action( 'scaip_shortcode', [ __CLASS__, 'create_placement_action' ] );
+		add_filter( 'newspack_ads_placements', [ __CLASS__, 'add_placements' ] );
+
+		// Deprecate sidebar.
+		remove_action( 'scaip_shortcode', 'scaip_shortcode_do_sidebar' );
+		add_filter( 'scaip_disable_sidebars', '__return_true' );
+		add_filter( 'newspack_ads_placement_data', [ __CLASS__, 'get_ad_unit_from_widget' ], 10, 3 );
```

Comment:
I think this is too heavy-handed. Some users might be used to managing their SCAIP settings in the widgets screen. The new UI should be an alternative one, not mutually exclusive with the default (widgets screen) UI.

Reply:
> The new UI should be an alternative one, not mutually exclusive with the default (widgets screen) UI.

The issue is that it is not an alternative UI, it is a different system entirely for handling the SCAIP placement.